### PR TITLE
Tgp 474/create remove nirms/niphl page

### PR DESCRIPTION
**Pull request by omotrium**

### CODE REVIEW

File: app/viewmodels/checkAnswers/HasNiphlsChangeSummary.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (besscerule) [inline app/viewmodels/checkAnswers/HasNiphlsChangeSummary.scala]: we don't need this page as it is never seen in a CYA page

File: app/viewmodels/checkAnswers/HasNirmsChangeSummary.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (besscerule) [inline app/viewmodels/checkAnswers/HasNirmsChangeSummary.scala]: see above

File: conf/app.routes [hunk 3]
```diff
@@ -209,3 +209,13 @@ GET         /update-record/:recordId/check-your-answers/trader-reference
 POST        /update-record/:recordId/check-your-answers/trader-reference                    controllers.CyaUpdateRecordController.onSubmitTraderReference(recordId: String)
 GET         /update-record/:recordId/check-your-answers/commodity-code                      controllers.CyaUpdateRecordController.onPageLoadCommodityCode(recordId: String)
 POST        /update-record/:recordId/check-your-answers/commodity-code                      controllers.CyaUpdateRecordController.onSubmitCommodityCode(recordId: String)
+
+GET        /update-profile/removing-nirms-number                                            controllers.HasNirmsChangeController.onPageLoad(mode: Mode = NormalMode)
+POST       /update-profile/removing-nirms-number                                            controllers.HasNirmsChangeController.onSubmit(mode: Mode = NormalMode)
+GET        /changeHasNirmsChange                                                            controllers.HasNirmsChangeController.onPageLoad(mode: Mode = CheckMode)
```
Comment (besscerule) [inline conf/app.routes]: in both cases we don't need the CheckMode version of these routes as we never view them from a CYA perspective we always go straight through the journey, so I would remove the CheckMode routes and then also remove the 'mode' parameters for both of the controllers as they are not needed

File: conf/messages.en [hunk 4]
```diff
@@ -451,3 +451,28 @@ hasGoodsDescriptionChange.p1 = Changing the goods description on this goods reco
...
+hasNirmsChange.no = No, remove my NIRMS details
+hasNirmsChange.warning = Warning
+hasNirmsChange.warningText = If you move goods under SPIMM and they are not eligible to, you could be liable for any incorrectly declared goods.
+
+hasNiphlsChange.title = hasNiphlsChange
+hasNiphlsChange.h1 = Removing NIPHL details
+hasNiphlsChange.p1 = If you remove your NIPHL details, the categorisation of goods you move under the Simplified Process for Internal Market Movements (SPIMM) could change.
+hasNiphlsChange.p2 = You will need to re-categorise those records to see if you can still move them under SPIMM.
+hasNiphlsChange.h2 = Are you sure you want to remove your NIPHL details?
+hasNiphlsChange.checkYourAnswersLabel = hasNiphlsChange
```
Comment (besscerule) [inline conf/messages.en]: this line isn't needed

File: conf/messages.en [hunk 5]
```diff
@@ -451,3 +451,28 @@ hasGoodsDescriptionChange.p1 = Changing the goods description on this goods reco
...
+hasNiphlsChange.error.required = Confirm if you want to remove your NIPHL
+hasNiphlsChange.change.hidden = HasNiphlsChange
```
Comment (besscerule) [inline conf/messages.en]: this line isn't needed

File: conf/messages.en [hunk 6]
```diff
@@ -451,3 +451,28 @@ hasGoodsDescriptionChange.p1 = Changing the goods description on this goods reco
 hasGoodsDescriptionChange.h2 = Do you want to change the goods description?
 hasGoodsDescriptionChange.error.required = Select if you want to change the goods description
 
+
+hasNirmsChange.title = hasNirmsChange
+hasNirmsChange.h1 = Removing NIRMS details
+hasNirmsChange.p1 = If you remove your NIRMS details, the categorisation of goods you move under the Simplified Process for Internal Market Movements (SPIMM) could change.
+hasNirmsChange.p2 = You will need to re-categorise those records to see if you can still move them under SPIMM.
+hasNirmsChange.h2 = Are you sure you want to remove your NIRMS details?
+hasNirmsChange.checkYourAnswersLabel = hasNirmsChange
```
Comment (besscerule) [inline conf/messages.en]: this line isn't needed

**Review by besscerule — CHANGES_REQUESTED** [review]
Virtually perfect :)

At some point we will need to update the data store with the new data - i think this will probably be on Submit of your two controllers if the user answers Yes - however I think that is part of the validate or navigation ticket so it doesn't need to be included in this ticket

File: app/views/HasNiphlsChangeView.scala.html [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+@(form: Form[_], mode: Mode)(implicit request: Request[_], messages: Messages)
+
+@layout(pageTitle = title(form, messages("hasNiphlsChange.title"))) {
+
+        <h1 class="govuk-heading-l">@messages("hasNiphlsChange.h1")</h1>
+
+            <p class="govuk-body">@messages("hasNiphlsChange.p1")</p>
+            <p class="govuk-body">@messages("hasNiphlsChange.p2")</p>
+
+        <div class="govuk-warning-text">
```
Comment (vinoocs) [inline app/views/HasNiphlsChangeView.scala.html]: can use the warning component as in the other PR https://github.com/hmrc/trader-goods-profiles-frontend/pull/214/files

**Review by besscerule — APPROVED** [review]

**Review by vinoocs — APPROVED** [review]

**Pull request merged**